2-mandatory/4-space-colonies.js

```diff
@@ -15,8 +15,17 @@
   
 */
 
-function getSettlers() {}
+function getSettlers(voyagers) {
 
+ let array=[];
+ let array1= [ array+"family" , "family"];
+ for(let familyName of voyagers){
+  if(familyName.split(' ').slice(-1).join(' ') && familyName[0] === "A"){
+  array.push(familyName)}
+}
+
+return array1;
```

Comment:
You could use filter here. (Note: I have not run this code so haven't checked syntax)
```js
function getSettlers(voyagers) {
  return  voyagers.filter(function(voyager) {
     // separate the name into an array of strings
    const names = voyager.split(' ');
    // check the name includes the string "family", the first word starts with "a"
    if (names.includes("family") && names[0].startsWith("a")) {
       return true
    } 
    return false
  })
}
```